### PR TITLE
Clarified what user data remains on logout

### PR DESCRIPTION
**Pull request by cwarnermm**
Documentation for: https://github.com/mattermost/docs/issues/6177

### CODE REVIEW

File: source/welcome/log-out.rst [hunk 1]
```diff
@@ -17,4 +17,22 @@ You can log out of Mattermost from your profile picture. Select **Log Out** to l
...
\ No newline at end of file
+        :alt: Log out of Mattermost from your profile picture.
+
+Frequently asked questions
+--------------------------
+
+What happens when I log out of Mattermost?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When you log out of Mattermost, all data related to your session is removed except a record in the app database of the server you accessed and general activity state information, such as the onboarding checklist. Your user data stored within the server database is deleted when you log out. If you were to delete the Mattermost desktop app, the most recent server URL and state data would also be deleted.
```
Comment (enahum) [inline source/welcome/log-out.rst:28]:
```suggestion
When you log out of Mattermost, all data related to your session is removed except a record in the app database of the server you accessed and general activity state information, such as the onboarding checklist. Your user data stored within the server database is deleted when you log out. If you were to delete the Mattermost desktop & mobile app, the most recent server URL and state data would also be deleted.
```

**Review by enahum — APPROVED** [review]

File: source/welcome/log-out.rst [hunk 2]
```diff
@@ -17,4 +17,22 @@ You can log out of Mattermost from your profile picture. Select **Log Out** to l
...
       .. image:: ../images/profile-log-out-mobile.png
         :width: 300
-        :alt: Log out of Mattermost from your profile picture.
\ No newline at end of file
+        :alt: Log out of Mattermost from your profile picture.
+
+Frequently asked questions
+--------------------------
+
+What happens when I log out of Mattermost?
```
Comment (hmhealey) [inline source/welcome/log-out.rst:25]: I'm pretty sure most of these only apply to mobile. I can't 100% speak for the desktop app, but most of these things don't exist in the web app or aren't stored in a way that lasts through a refresh

File: source/welcome/log-out.rst [hunk 3]
```diff
@@ -17,4 +17,22 @@ You can log out of Mattermost from your profile picture. Select **Log Out** to l
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When you log out of Mattermost, all data related to your session is removed except a record in the app database of the server you accessed and general activity state information, such as the onboarding checklist. Your user data stored within the server database is deleted when you log out. If you were to delete the Mattermost desktop and the mobile app, the most recent server URL and state data would also be deleted.
+
+When you log out, the following additional data is also deleted:
+- On MacOS, your user credentials for this server from the keychain.
```
Comment (hmhealey) [inline source/welcome/log-out.rst:31]: In particular, I don't think we explicitly interact with the keychain even from the desktop app

Comment (devinbinnie) [reply source/welcome/log-out.rst:31]: This is correct, there's no keychain added for web or desktop afaik.

**Review by devinbinnie — APPROVED** [review]

**Pull request merged**